inbm/cloudadapter-agent/cloudadapter/cloud/client/connections/mqtt_connection.py

```diff
@@ -95,8 +95,10 @@ def _set_proxy(self, config: ProxyConfig) -> None:
         @param config: (ProxyConfig) Config to use
         """
         if config.endpoint:
-            socks.set_default_proxy(socks.PROXY_TYPE_HTTP, *config.endpoint)
-            socket.socket = socks.socksocket  # type: ignore
+            proxy_addr, proxy_port = config.endpoint
+            self._client.proxy_set(proxy_type=socks.HTTP,
+                                   proxy_addr=proxy_addr,
+                                   proxy_port=proxy_port)
```

Comment:
**<h2>$\color{#FFA500}Improvements: CodeReadability$</h2>**
**Suggestion:** Replace the magic number for proxy_type with a constant that defines what the number represents for better code readability and maintainability.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        HTTP_PROXY_TYPE = socks.HTTP
        if config.endpoint:
            proxy_addr, proxy_port = config.endpoint
            self._client.proxy_set(proxy_type=HTTP_PROXY_TYPE,
                                   proxy_addr=proxy_addr,
                                   proxy_port=proxy_port)
```

Reply:
that's not a magic number, that's a defined constant already.